Response/index.ts

```diff
@@ -3,37 +3,47 @@ import * as Serializer from "../Serializer"
 import { Header as ResponseHeader } from "./Header"
 import { Like as ResponseLike } from "./Like"
 
-export interface Response {
+export interface HttpResponse {
 	readonly status: number
 	readonly header: Readonly<ResponseHeader>
 	readonly body?: any | Promise<any>
+	readonly webSocket?: WebSocket | null
 }
 
-export namespace Response {
-	export function is(value: any | Response): value is Response {
+export namespace HttpResponse {
+	export function is(value: any | HttpResponse): value is HttpResponse {
 		return (
 			typeof value == "object" &&
-			Object.keys(value).every(key => ["status", "header", "body"].some(k => k == key)) &&
+			Object.keys(value).every(key => ["status", "header", "body", "webSocket"].some(k => k == key)) &&
 			(value.status == undefined || typeof value.status == "number") &&
-			(value.header == undefined || ResponseHeader.is(value.header))
+			(value.header == undefined || ResponseHeader.is(value.header)) &&
+			(value.webSocket == undefined || value.webSocket instanceof WebSocket)
 		)
 	}
-	export async function to(request: Response): Promise<globalThis.Response> {
-		return new globalThis.Response(await Serializer.serialize(await request.body, request.header.contentType), {
+	export async function to(request: HttpResponse): Promise<Response> {
+		return new Response(await Serializer.serialize(await request.body, request.header.contentType), {
 			status: request.status,
-			headers: new globalThis.Headers(ResponseHeader.to(request.header) as Record<string, string>),
-		})
+			headers: new Headers(ResponseHeader.to(request.header) as Record<string, string>),
+			webSocket: request.webSocket,
+		} as ResponseInit & Partial<{ webSocket: WebSocket | null }>)
 	}
-	export function from(response: globalThis.Response): Response {
+	export function from(response: Response & Partial<{ webSocket?: WebSocket | null }>): HttpResponse {
 		return {
 			status: response.status,
 			header: ResponseHeader.from(response.headers),
-			body: Parser.parse(response),
+			body: response.status == 101 ? null : Parser.parse(response),
```

Comment:
Don't use `null` use `undefined`.

Reply:
I removed the null referenced here and from as many places as I could but the remaining ones used for typing are required to be null to be compatible with the [cloudflare response type](https://github.com/cloudflare/workers-types/blob/master/index.d.ts#L1472). It still builds locally for me with all of them not being null but it produces TS2345 build errors. So I think its better to leave the remaining nulls to not produce errors.